service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -179,36 +185,78 @@ private String prepareReportLevelNarrativeStatements(DiagnosticReport diagnostic
 
             reportLevelNarrativeStatements.append(statusNarrativeStatement);
         }
 
         buildNarrativeStatementForMissingResults(diagnosticReport, reportLevelNarrativeStatements);
+        buildNarrativeStatementForParticipants(diagnosticReport, reportLevelNarrativeStatements);
 
         return reportLevelNarrativeStatements.toString();
     }
 
     private void buildNarrativeStatementForMissingResults(DiagnosticReport diagnosticReport, StringBuilder reportLevelNarrativeStatements) {
         if (reportLevelNarrativeStatements.length() == 0 && !diagnosticReport.hasResult()) {
             String narrativeStatementFromCodedDiagnosis = buildNarrativeStatementForDiagnosticReport(
-                diagnosticReport, CommentType.AGGREGATE_COMMENT_SET.getCode(), "EMPTY REPORT"
+                diagnosticReport.getIssuedElement(), CommentType.AGGREGATE_COMMENT_SET.getCode(), "EMPTY REPORT"
             );
             reportLevelNarrativeStatements.append(narrativeStatementFromCodedDiagnosis);
         }
     }
 
-    private String buildNarrativeStatementForDiagnosticReport(DiagnosticReport diagnosticReport, String commentType, String comment) {
+    private void buildNarrativeStatementForParticipants(DiagnosticReport diagnosticReport, StringBuilder reportLevelNarrativeStatements) {
+        if (diagnosticReport.hasPerformer()) {
+            var humanNames = buildListOfHumanReadableNames(diagnosticReport.getPerformer());
+            String performerNarrativeStatement = buildNarrativeStatementForDiagnosticReport(
+                diagnosticReport.getIssuedElement(), CommentType.AGGREGATE_COMMENT_SET.getCode(),
+                PREPENDED_TEXT_FOR_PARTICIPANTS + humanNames
+            );
+            reportLevelNarrativeStatements.append(performerNarrativeStatement);
+        }
+    }
+
+    private String buildNarrativeStatementForDiagnosticReport(InstantType issuedElement, String commentType, String comment) {
         var narrativeStatementTemplateParameters = NarrativeStatementTemplateParameters.builder()
             .narrativeStatementId(randomIdGeneratorService.createNewId())
             .commentType(commentType)
-            .commentDate(DateFormatUtil.toHl7Format(diagnosticReport.getIssuedElement()))
+            .commentDate(DateFormatUtil.toHl7Format(issuedElement))
             .comment(comment)
-            .availabilityTimeElement(StatementTimeMappingUtils.prepareAvailabilityTime(diagnosticReport.getIssuedElement()));
+            .availabilityTimeElement(StatementTimeMappingUtils.prepareAvailabilityTime(issuedElement));
 
         return TemplateUtils.fillTemplate(
             NARRATIVE_STATEMENT_TEMPLATE,
             narrativeStatementTemplateParameters.build()
         );
     }
+
+    private String buildListOfHumanReadableNames(List<DiagnosticReport.DiagnosticReportPerformerComponent> performers) {
+        return
+            performers.stream()
```

Comment:
A bit strange formatting

Reply:
Not sure what happened there, will fix it